app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -79,6 +79,32 @@ def metrics_connect
                                      authentication_password('default'))
     end
 
+    def run_generic_operation(operation, path)
+      require 'hawkular_all'
+      host_port = "#{hostname}:#{port}"
+      client = ::Hawkular::Operations::OperationsClient.new(:host        => host_port,
+                                                            :credentials => {
+                                                              :username => authentication_userid('default'),
+                                                              :password => authentication_password('default')
+                                                            })
+      the_operation = {
+        :operationName => operation,
+        :resourcePath  => path.to_s
+      }
+
+      actual_data = {}
+      client.invoke_generic_operation(the_operation) do |on|
+        on.success do |data|
+          _log.debug "WS Success #{data}"
```

Comment:
Websocket.
I guess that part needs an overhaul and also (in the future) a way to return the result to the caller. But right now this result is not evaluated anyway.